arangod/Agency/Agent.cpp

```diff
@@ -1490,14 +1511,13 @@ Store const& Agent::spearhead() const { return _spearhead; }
 /// a const reference
 /// the caller has to make sure the lock is actually held
 Store const& Agent::readDB() const { 
-  _ioLock.assertLockedByCurrentThread();
+  MUTEX_LOCKER(oLocker, _oLock);
   return _readDB; 
```

Comment:
This is returning a reference to the entire object.  I do not see anything that the lock can protect.  In fact, I think the assert() was better and just needs to change variables.